exec/java-exec/src/main/java/org/apache/drill/exec/store/sys/FunctionsIterator.java

```diff
@@ -0,0 +1,116 @@
+ */
+package org.apache.drill.exec.store.sys;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.drill.exec.expr.fn.DrillFuncHolder;
+import org.apache.drill.exec.expr.fn.FunctionImplementationRegistry;
+import org.apache.drill.exec.expr.fn.registry.FunctionHolder;
+import org.apache.drill.exec.expr.fn.registry.LocalFunctionRegistry;
+import org.apache.drill.exec.ops.ExecutorFragmentContext;
+import org.apache.drill.exec.store.pojo.NonNullable;
+
+/**
+ * List functions as a System Table
+ */
+public class FunctionsIterator implements Iterator<Object> {
+  //private static final org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(FunctionsIterator.class);
+
+  private Iterator<FunctionInfo> sortedIterator;
+
+  public FunctionsIterator(ExecutorFragmentContext context) {
+    Map<String, FunctionInfo> functionMap = new HashMap<String, FunctionInfo>();
+    //Access Registry for function list
+    FunctionImplementationRegistry funcImplRegistry = (FunctionImplementationRegistry) context.getFunctionRegistry();
+    Map<String, List<FunctionHolder>>  jarFunctionListMap = funcImplRegistry.getAllFunctionsHoldersByJar();
+
+    //Step 1: Load Built-In
+    for (FunctionHolder builtInEntry : jarFunctionListMap.get(LocalFunctionRegistry.BUILT_IN)) {
+      populateFunctionMap(functionMap, LocalFunctionRegistry.BUILT_IN, builtInEntry.getHolder());
+    }
+    //Step 2: Load UDFs
+    for (String jarName : new ArrayList<>(jarFunctionListMap.keySet())) {
+      if (!jarName.equals(LocalFunctionRegistry.BUILT_IN)) {
+        for (FunctionHolder udfEntry : jarFunctionListMap.get(jarName)) {
+          populateFunctionMap(functionMap, jarName, udfEntry.getHolder());
+        }
+      }
+    }
+
+    List<FunctionInfo> functionList = new ArrayList<FunctionsIterator.FunctionInfo>(functionMap.values());
+    functionList.sort(new Comparator<FunctionInfo>() {
+      @Override
+      public int compare(FunctionInfo o1, FunctionInfo o2) {
+        int result = o1.name.compareTo(o2.name);
+        if (result == 0) {
+          result = o1.signature.compareTo(o2.signature);
+        }
+        if (result == 0) {
+          return o1.returnType.compareTo(o2.returnType);
+        }
+        return result;
+      }
+    });
+
+    sortedIterator = functionList.iterator();
+  }
+
+  //Populate for give Jar-FunctionHolder Entry
+  private void populateFunctionMap(Map<String, FunctionInfo> functionMap, String jarName, DrillFuncHolder dfh) {
+    String registeredNames[] = dfh.getRegisteredNames();
+    String signature = dfh.getInputParameters();
+    String returnType = dfh.getReturnType().getMinorType().toString();
+    for (String name : registeredNames) {
+      //Generate a uniqueKey to allow for fnName#fnSignature
+      String funcSignatureKey = new StringBuilder(64).append(name).append('#').append(signature).toString();

```

Comment:
Please explain the purpose of this unique key.

Reply:
The original loading of functions from `built-in` captures functions from all the sources. When iterating over the non-`built-in` sources, the key helps replace `FunctionInfo` instances that are wrongly shown as sourced from `built-in`

Reply:
Put it in the comments

Reply:
Why `StringBuilder` capacity is 64?